central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -54,23 +87,152 @@ func (m *managerImpl) ProcessComplianceOperatorInfo(ctx context.Context, complia
 	return m.integrationDS.UpdateComplianceIntegration(ctx, complianceIntegration)
 }
 
-func (m *managerImpl) ProcessScanRequest(_ context.Context, _ interface{}) error {
-	//TODO:
-	// 1. Upsert to database
-	// 2. Push request to Sensor
-	panic("implement me")
+// ProcessScanRequest processes a request to apply a compliance scan configuration to one or more Sensors.
+func (m *managerImpl) ProcessScanRequest(ctx context.Context, scanRequest *storage.ComplianceOperatorScanConfigurationV2, clusters []string) (*storage.ComplianceOperatorScanConfigurationV2, error) {
+	if !features.ComplianceEnhancements.Enabled() {
+		return nil, nil
+	}
+
+	// Convert and validate schedule
+	// TODO(ROX-19818):  remove default schedule
+	// For MVP a schedule is required.  If one is not present use daily at midnight.  As it evolves
+	// it may be as simple as no schedule means a one time scan.
+	cron := defaultScanSchedule
+	var err error
+	if scanRequest.GetSchedule() != nil {
+		cron, err = schedule.ConvertToCronTab(scanRequest.GetSchedule())
+		if err != nil {
+			log.Error(err)
+			return nil, errors.Errorf("Unable to convert schedule for scan configuration named %q to cron.", scanRequest.GetScanName())
+		}
+		cronValidator := gronx.New()
+		if !cronValidator.IsValid(cron) {
+			log.Error(err)
+			return nil, errors.Errorf("Schedule for scan configuration named %q is invalid.", scanRequest.GetScanName())
+		}
+	}
+
+	// Check if scan configuration already exists.
+	found, err := m.scanSettingDS.GetScanConfigurationExists(ctx, scanRequest.GetScanName())
+	if err != nil {
+		log.Error(err)
+		return nil, errors.Wrapf(err, "Unable to create scan configuration named %q.", scanRequest.GetScanName())
+	}
+	if found {
+		return nil, errors.Errorf("Scan configuration named %q already exists.", scanRequest.GetScanName())
+	}
+
+	scanRequest.Id = uuid.NewV4().String()
+	scanRequest.CreatedTime = types.TimestampNow()
+	err = m.scanSettingDS.UpsertScanConfiguration(ctx, scanRequest)
+	if err != nil {
+		log.Error(err)
+		return nil, errors.Errorf("Unable to save scan configuration named %q.", scanRequest.GetScanName())
+	}
+
+	var profiles []string
+	for _, profile := range scanRequest.GetProfiles() {
+		profiles = append(profiles, profile.GetProfileName())
+	}
+
+	m.requestsLock.Lock()
+	defer m.requestsLock.Unlock()
```

Comment:
You should not hold lock around a grpc call. You only need to protect the updates the `requests` map.